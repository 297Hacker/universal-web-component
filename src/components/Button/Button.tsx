import React, {useRef, useEffect} from 'react';
import styled from 'styled-components';

type ButtonProps = {
    addClass?: string
    myClass?: string
    name?: string
    children?: string
    value?: string
    kind?: string[]
    onClick?: any
    onChange?: any
    type?: string
}

const MainButton = styled.button`
  background-color: #82BFE0;
  border-radius: 2px;
  border: none;
  color: #FFF;
  margin: 0 10px;
  padding: 10px 40px;
    &:hover {
        background-color: #000;
        cursor: pointer;
    }
`;

const Span = styled.span`
    display: none;
    
    &.view {
    display: block;
    }
`;

const Button: React.FC<ButtonProps> = (props) => {
    const focusButton:any = useRef(null);

    useEffect(() => {
        console.log('component is being rerendered');
    }, []);

    return <>
            <MainButton ref={focusButton}
                        className={props.myClass}
                        name={props.name}
                        value={props.value}
                        onClick={props.onClick}
                        onChange={props.onChange}>
                {props.children}
            </MainButton>
            <Span className={props.myClass}>view content when button is clicked</Span>
    </>
};

Button.defaultProps = {
    myClass: 'button',
    name: 'button',
    value: 'Click',
    children: 'Click button'
};

export default Button;