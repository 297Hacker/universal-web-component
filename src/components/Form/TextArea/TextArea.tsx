import React from 'react';
import styled from 'styled-components';

type textAreaProps = {
    totalRows: number
    children: number & string
}

const TextAreaContent = styled.textarea`
    height: 100px;
    border: 1px solid #eee;
    border-radius: 3px;
    font-size: 16px;
    margin: 10px 0;
    padding: 10px 5px;
    outline: none;
    resize: none;
    width: 250px;
`;

const TextArea: React.FC<textAreaProps> = (props) => {
    return <React.Fragment>
        <TextAreaContent rows={props.totalRows}>{props.children}</TextAreaContent>
    </React.Fragment>;
};

export default TextArea;
