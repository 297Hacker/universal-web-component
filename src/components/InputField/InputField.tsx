import React, {useState} from 'react';
import styled from 'styled-components';

type InputFieldType = {
    name: string
    type: string
    input?: string
    placeholder?: string
    maxValue: number
    value?: string
}

const Input = styled.input`
    border: 1px solid #eee;
    border-radius: 3px;
    font-size: 16px;
    margin: 10px 10px 10px 0;
    padding: 10px 5px;
    outline: none;
    width: 250px;
`;

const InputField: React.FC<InputFieldType> = (props) => {
    const [input, setInput] = useState();
    const lengthMax: number = props.maxValue;
    const lengthWarning: number = 15;
    let warningText: {} = "";

    const handleChange = (e: any) => {
        setInput(e.target.value);
        if (e.target.value.length > lengthWarning && e.target.value.length <= lengthMax) {
            console.log('length check');
            warningText = <span>max text length is almost reached</span>;
        } else {
            console.log('warning check');
            warningText = <span> </span>;
        }
    };

    const handleBlur = () => {
    };

    return <>
        <Input maxLength={props.maxValue} name={props.name} value={input}
               type={props.type} placeholder={props.placeholder}
               onChange={e => handleChange(e)} onBlur={() => handleBlur()}
        />
        {warningText}
    </>;
};

InputField.defaultProps = {
    name: 'input name',
    type: 'name',
    maxValue: 20,
    placeholder: 'Name'
};

export default InputField;
