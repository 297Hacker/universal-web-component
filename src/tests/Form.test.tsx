import React from 'react';
import {shallow} from 'enzyme';
import ContactForm from '../components/Form/Contact/ContactForm';
import InputField from '../components/InputField/InputField';
import TextArea from '../components/Form/TextArea/TextArea';
import Button from '../components/Button/Button';

describe('<ContactForm/>', () => {
    it("Contact form has two inputfield,with a Textare and a button", () => {
        const component = shallow(<ContactForm/>);
        expect(component.find(InputField)).toHaveLength(2);
        expect(component.find(TextArea)).toHaveLength(1);
        expect(component.find(Button)).toHaveLength(1);
        expect(component).toMatchSnapshot();
    });
});

