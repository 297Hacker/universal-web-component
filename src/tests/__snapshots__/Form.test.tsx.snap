// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactForm/> Contact form has two inputfield,with a Textare and a button 1`] = `
<Fragment>
  <styled.form
    className="contact-form"
  >
    <fieldset>
      <legend />
      <div>
        <label>
          Name:
        </label>
        <InputField
          maxValue={50}
          name="name"
          placeholder="Name"
          type="name"
        />
      </div>
      <div>
        <label>
          Email:
        </label>
        <InputField
          maxValue={25}
          name="email"
          placeholder="Email"
          type="email"
        />
      </div>
      <div>
        <label>
          Message:
        </label>
        <TextArea />
      </div>
      <Component
        onClick={[Function]}
        type="button"
      >
        Send
      </Component>
      <styled.span>
        this is viewed when clicked
      </styled.span>
    </fieldset>
  </styled.form>
</Fragment>
`;

exports[`<ContactForm/> menu has two link component 1`] = `
<Fragment>
  <styled.form
    className="contact-form"
  >
    <fieldset>
      <legend />
      <div>
        <label>
          Name:
        </label>
        <InputField
          maxValue={50}
          name="name"
          placeholder="Name"
          type="name"
        />
      </div>
      <div>
        <label>
          Email:
        </label>
        <InputField
          maxValue={25}
          name="email"
          placeholder="Email"
          type="email"
        />
      </div>
      <div>
        <label>
          Message:
        </label>
        <TextArea />
      </div>
      <Component
        onClick={[Function]}
        type="button"
      >
        Send
      </Component>
      <styled.span>
        this is viewed when clicked
      </styled.span>
    </fieldset>
  </styled.form>
</Fragment>
`;
